### PR TITLE
Add basic integration test

### PR DESCRIPTION
**Pull request by barisbalic**
Most basic case, PRing to start discussion on where we put files, and see what opinions arise.

Also, defining a mailer seemed like overkill, but happy to do it if you disagree.

### CODE REVIEW

File: test/integration/adapters/mailgun_test.exs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+defmodule Swoosh.Integration.MailgunAdapterTest do
+  use ExUnit.Case, async: true
+
+  alias Swoosh.Adapters.Mailgun
+  import Swoosh.Email
+
+  @tag external: true
```
Comment (stevedomin) [inline test/integration/adapters/mailgun_test.exs]: Is that for the following test case or the whole module? If the latter we should use `@moduletags` I think

Comment (barisbalic) [inline test/integration/adapters/mailgun_test.exs]: Not seen `@moduletags` but judging by the ExUnit documentation `@tags` is for the module...

File: test/integration/adapters/mailgun_test.exs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    email =
+      %Swoosh.Email{}
+      |> from("Mailgun Test <mailgun@#{sandbox_domain}>")
+      |> to("leafybasil@gmail.com")
+      |> subject("Test Email")
+      |> text_body("Hey, just testing!")
+      |> html_body("<h1>Hey, just <i>testing!</i></h1>")
+
+    {status, response} = Swoosh.Adapters.Mailgun.deliver(email, config)
```
Comment (stevedomin) [inline test/integration/adapters/mailgun_test.exs]: why are you not asserting directly that statement?

```elixir
assert {:ok, response} = Swoosh.Adapters.Mailgun.deliver(email, config)
```

File: test/integration/adapters/mailgun_test.exs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  @tag external: true
+
+  test "Mailgun Adapter" do
+    sandbox_domain = "/sandboxac405b2c12824f69b7cea45fb8cb6e10.mailgun.org"
+
+    config = %{domain: sandbox_domain, api_key: System.get_env("MAILGUN_API_KEY")}
+
+    email =
+      %Swoosh.Email{}
```
Comment (stevedomin) [inline test/integration/adapters/mailgun_test.exs]: should we add `cc`, `bcc` and `reply_to` as well?

File: test/integration/adapters/mailgun_test.exs [hunk 4]
```diff
@@ -0,0 +1,26 @@
+defmodule Swoosh.Integration.MailgunAdapterTest do
+  use ExUnit.Case, async: true
+
+  alias Swoosh.Adapters.Mailgun
+  import Swoosh.Email
+
+  @tag external: true
+
+  test "Mailgun Adapter" do
+    sandbox_domain = "/sandboxac405b2c12824f69b7cea45fb8cb6e10.mailgun.org"
```
Comment (stevedomin) [inline test/integration/adapters/mailgun_test.exs]: I'd make that a env variable, that way other contributors can use it easily too

File: test/integration/adapters/mailgun_test.exs (hunk 1 above)
Comment (barisbalic) [inline test/integration/adapters/mailgun_test.exs]: Confirmed, it counts for the module.

File: test/integration/adapters/mailgun_test.exs (hunk 2 above)
Comment (barisbalic) [inline test/integration/adapters/mailgun_test.exs]: No reason, happy to do it!

File: test/integration/adapters/mailgun_test.exs (hunk 3 above)
Comment (barisbalic) [inline test/integration/adapters/mailgun_test.exs]: :100:

File: test/integration/adapters/mailgun_test.exs (hunk 4 above)
Comment (barisbalic) [inline test/integration/adapters/mailgun_test.exs]: Yeah intended to, just hadn't done yet.

File: test/integration/adapters/mailgun_test.exs (hunk 1 above)
Comment (stevedomin) [inline test/integration/adapters/mailgun_test.exs]: Meant `@moduletag`

`@tag` is definitely for individual tests => http://elixir-lang.org/docs/stable/ex_unit/ExUnit.Case.html

File: test/test_helper.exs [hunk 5]
```diff
@@ -1,3 +1,4 @@
 ExUnit.start()
+ExUnit.configure exclude: [external: true]
```
Comment (stevedomin) [inline test/test_helper.exs]: Minor thing but this can be `ExUnit.configure(exclude: :external)`

**Comment by barisbalic** [comment]
@stevedomin this definitely needs rebasing against master, but I didn't have credentials to run/test Sendgrid or SMTP to hand, could you please run them.  I will rebase and fix up any feedback tonight.

**Comment by barisbalic** [comment]
Rebased, because I need to stop putting off things that take a few seconds.

**Pull request merged**